### PR TITLE
Conste struct devices for Zephyr 2.4.

### PR DESCRIPTION
**Pull request by daniel-connectedenergy**
Zephyr 2.4 support.

### CODE REVIEW

File: src/SIM800-AT-Zephyr.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ int GPRS::request_data(void)
     return MODEM_RESPONSE_OK;
 }
 
-void read_resp(void* user_data)
+void read_resp(const device *dev, void* user_data)
```
Comment (martinjaeger) [inline src/SIM800-AT-Zephyr.cpp:48]: Missing `struct`?

**Review by martinjaeger — CHANGES_REQUESTED** [review]

**Comment by azeemshatp** [comment]
So, are we going to leave this branch in parallel?

**Comment by daniel-connectedenergy** [comment]
I need to find some time to squeeze in a day of work to make sure this works properly - this means also fixing the CI.

**Review by martinjaeger — APPROVED** [review]

**Comment by martinjaeger** [comment]
From my point of view, we can make master compatible to Zephyr 2.4 and if necessary create a 2.3 branch to track from SB if there is any reason for not switching to v2.3 in SB.

If anyone disagrees, please leave comments here.

**Comment by JordanSilverman** [comment]
It's fine by me. @azeemshatp maybe you could have a look at upgrading SB to zephyr 2.4? I don't know what the process is.

**Comment by martinjaeger** [comment]
Ok, if @azeemshatp agrees I will merge this so that I can create the other PR in cloudsolar firmware.

**Comment by daniel-connectedenergy** [comment]
I concur - we should move to Zephyr 2.4.

**Pull request merged**